service/storage_proxy.cc

```diff
@@ -3076,7 +3092,7 @@ class digest_read_resolver : public abstract_read_resolver {
             return true;
         }
         auto& first = *_digest_results.begin();
-        return std::find_if(_digest_results.begin() + 1, _digest_results.end(), [&first] (query::result_digest digest) { return digest != first; }) == _digest_results.end();
+        return std::find_if(_digest_results.begin() + 1, _digest_results.end(), [&first] (digest_and_last_pos digest) { return digest.digest != first.digest; }) == _digest_results.end();
```

Comment:
This is introducing an allocation. query::result_digest is just 16 bytes and can be passed in registers, but digest_and_last_pos is a complex type. Better to pass by reference.

Reply:
right, will do